File: go/wam/5308.go
const WamSearchActionEvent = 5308 // channel: regular

type SearchActionEventEvent struct {
	BizSearchCount int64 `wam:"biz_search_count"` // field 1
	SearchAction SEARCHACTIONTYPE `wam:"search_action"` // field 2
	SearchActionEntryPoint SEARCHACTIONENTRYPOINTTYPE `wam:"search_action_entry_point"` // field 3
	SearchAiSuggestionCount int64 `wam:"search_ai_suggestion_count"` // field 4 (search_ai_suggestion_count > 0; search_ai_suggestion_count == 1)
	SearchChatsCount int64 `wam:"search_chats_count"` // field 5
	SearchContactsCount int64 `wam:"search_contacts_count"` // field 6
	SearchFilterCount int64 `wam:"search_filter_count"` // field 7
	SearchGroupsCount int64 `wam:"search_groups_count"` // field 8
	SearchMessagesCount int64 `wam:"search_messages_count"` // field 9
	SelectedItemRank int64 `wam:"selected_item_rank"` // field 10
	ResultPageShown bool `wam:"result_page_shown"` // field 11
}

File: rust/wam/5308.rs
pub const WAM_SEARCH_ACTION_EVENT: u32 = 5308;

#[derive(Debug, Default)]
pub struct SearchActionEventEvent {
    /// field 1, wire `biz_search_count`
    pub biz_search_count: Option<i64>,
    /// field 2, wire `search_action`
    pub search_action: Option<SEARCHACTIONTYPE>,
    /// field 3, wire `search_action_entry_point`
    pub search_action_entry_point: Option<SEARCHACTIONENTRYPOINTTYPE>,
    /// field 4, wire `search_ai_suggestion_count` (search_ai_suggestion_count > 0; search_ai_suggestion_count == 1)
    pub search_ai_suggestion_count: Option<i64>,
    /// field 5, wire `search_chats_count`
    pub search_chats_count: Option<i64>,
    /// field 6, wire `search_contacts_count`
    pub search_contacts_count: Option<i64>,
    /// field 7, wire `search_filter_count`
    pub search_filter_count: Option<i64>,
    /// field 8, wire `search_groups_count`
    pub search_groups_count: Option<i64>,
    /// field 9, wire `search_messages_count`
    pub search_messages_count: Option<i64>,
    /// field 10, wire `selected_item_rank`
    pub selected_item_rank: Option<i64>,
    /// field 11, wire `result_page_shown`
    pub result_page_shown: Option<bool>,
}